editor/plugins/shader_editor_plugin.cpp

```diff
@@ -561,41 +826,132 @@ void ShaderEditor::edit(const Ref<Shader> &p_shader) {
 	}
 
 	shader = p_shader;
+	shader_inc = nullptr;
+
+	// Create shader dependencies and update tree.
+	shader_rolling_code.clear();
+	shader_dependencies.populate(shader);
+	_update_shader_dependency_tree();
+
+	// Show and hide dep tree based on amount of deps. If 1 dep, don't show.
+	if (shader_dependencies.size() > 1) {
+		shader_dependency_tree->show();
+	} else {
+		shader_dependency_tree->hide();
+	}
+
+	current_path = shader->get_path();
+	shader_rolling_code[current_path] = p_shader->get_code();
 
 	shader_editor->set_edited_shader(p_shader);
+}
+
+void ShaderEditor::edit(const Ref<ShaderInclude> &p_shader_inc) {
+	if (p_shader_inc.is_null()) {
+		return;
+	}
+
+	if (shader_inc == p_shader_inc) {
+		return;
+	}
+
+	shader_inc = p_shader_inc;
+	shader = nullptr;
```

Comment:
I manually changed this to ``shader= Ref<Shader>();`` Im not sure if it is correct to do so.